doc/track_2_migration_guidance.md

```diff
@@ -0,0 +1,115 @@
+
+This document summarizes typical issues and solutions when adopting Track 2 SDK in Azure CLI.
+
+Example PRs:
+1. [Compute PR](https://github.com/Azure/azure-cli/pull/15750)
+2. [Network PR](https://github.com/Azure/azure-cli/pull/16245)
+
+Below is a list of typical issues.
+
+1. Long running operation function name change
+2. Property name change
+3. Class name change
+4. Error type change
+5. No enum type
+6. Class hierarchy change
+7. Obtaining Subscription
+8. Multi-API support
+9. Fixing mocked object
+10. Modifying patch_models.py to include missing packages
+11. Missing x-ms-authorization-auxiliary
+
+### Long running operation function name change
+
+Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation.
+
+### Property name change
+
+Some of property names change in Track 2 SDK.
+
+Examples:
+
+```
+hyperVgeneration -> hyperVGeneration
+disk_mbps_read_only -> disk_m_bps_read_only
+disk_mbps_read_write -> disk_m_bps_read_write
+virtual_machine_extension_type -> type_properties_type
+type1 -> type_properties_type
+instance_ids -> vm_instance_i_ds
+diskMbpsReadWrite -> diskMBpsReadWrite
+```
+
+Some changes are unnecessary, even wrong in English. I opened [Azure/autorest.python#850](https://github.com/Azure/autorest.python/issues/850) to track this problem.
+
+### Class name change
+
+Some of class names change in Track 2 SDK.
+
+Examples:
+
+```
+VirtualMachineIdentityUserAssignedIdentitiesValue -> UserAssignedIdentitiesValue
+```
+
+### Error type change
+
+Error type changes in Track2 SDK.
+
+Examples:
+
+```
+CloudError -> azure.core.exceptions.ResourceNotFoundError
+```
+
+### No enum type
+
+Track 2 SDK removes enum type and adopts string type instead. It loses the validation on values. Anyway, do not use `obj.value` any longer. Just use `obj`.
+
+### Class hierarchy change
+
+The class hierarchy may change in Track 2 SDK. Some properties are not flattened. They are wrapped in classes. 
+
+Examples:
+
+In VMSS `begin_update_instances`, a new type `VirtualMachineScaleSetVMInstanceRequiredIDs` is added.
+
+In DiskAccess `begin_create_or_update`, location and tags are moved to a nested structure `DiskAccess`, `disk_access = DiskAccess(location=location, tags=tags)`
+
+### Obtaining Subscription
```

Comment:
Better to use base form of verbs.

Reply:
http://www.davidnicholson.it/language/on-ing-forms-in-titles.html
I prefer gerund.